src/NuGet.Services.BasicSearch/WebRole.cs

```diff
@@ -68,20 +73,24 @@ public override void OnStop()
                     Thread.Sleep(TimeSpan.FromSeconds(1));
                 }
             }
+
+            _httpClient = null;
         }
 
-        private void MakeRequest(string url)
+        private async Task MakeRequestAsync(HttpClient httpClient, string url)
         {
+            if (httpClient == null)
+            {
+                return;
+            }
+
             try
             {
-                var request = (HttpWebRequest)WebRequest.Create(url);
```

Comment:
🤢

Reply:
If you think this is bad, wait until you see the Gallery functional tests! 🦀